### PR TITLE
Adoc Attributes

### PR DESCRIPTION
**Pull request by hkp-wcd**
Gets attributes from current adoc document and all its included files before doing the id pattern checks

### CODE REVIEW

File: lib/base_extension.rb [hunk 1]
```diff
@@ -53,19 +53,23 @@ def create_error(msg:, location: nil, extras: nil)
       }
     end
 
-    ##
```
Comment (ldz-w) [inline lib/base_extension.rb]: @hkp-wcd Should be this doc string, not the old one.

File: lib/extensions.d/link_checker.rb [hunk 2]
```diff
@@ -27,20 +27,16 @@ def self.format_net_exception(exc, link)
   #
   # Check links and detect whether a link is dead, has moved, cannot be reached, etc.
   class LinkChecker < BaseExtension
-    ##
```
Comment (ldz-w) [inline lib/extensions.d/link_checker.rb]: @hkp-wcd doc string missing

File: lib/extensions.d/pattern_blacklist.rb [hunk 3]
```diff
@@ -9,13 +9,10 @@ module Toolchain
...
   # Checks the text against a predefined list of patterns which are not allowed.
   class PatternBlacklist < BaseExtension
-    ##
-    # Run the Pattern tests on the given document (+document+, +original+).
-    # Illegal patterns are loaded from +blacklist_file+.
-    #
-    # Returns a list of errors (can be empty).
-    #
-    def run(document, original, blacklist_file = '../blacklist.txt')
+    def run(adoc, blacklist_file = '../blacklist.txt')
```
Comment (ldz-w) [inline lib/extensions.d/pattern_blacklist.rb]: @hkp-wcd doc string missing

**Comment by hkp-wcd** [comment]
Obsolete. New branch Attributes.

**Pull request closed without merging**